getting-help/questions.rst

```diff
@@ -0,0 +1,124 @@
+      previous post. We realize the volume of posts can be overwhelming,
+      but you should start by using the discussion board’s search
+      functionality to see if it brings up any relevant posts. For
+      example, suppose you are failing a specific test in a programming assignment;
+      you could search just for that test’s name to see if any
+      other students have encountered that same error (and, if you’re
+      lucky, an instructor/TA will have already answered it).
+   -  **Make sure you’re going to ask an actual question**. Make sure
+      you are describing a specific issue you’re encountering, and why you’re
+      stuck on it (e.g., you are not getting the expected result, the
+      tests are failing in a way you do not understand, etc.). Writing a
+      post that says “I can’t get Task 4 to work, I’ve pushed my code.
+      Please look at it.” is not a question: it is a request for someone
+      to debug your code for you. Try instead to describe the exact
+      reason why you're stuck (we provide concrete suggestions on how to
+      do this below). If you're struggling to do so, you may want to
+      go to :ref:`office hours <getting-help_office-hours>` to get help
+      more interactively.
+
+- **Public vs private questions**. Discussion boards often have
+  mechanisms that allow you to ask private questions, which will be
+  seen only by the instructors and teaching assistants. Typically, this
+  mechanism should be used only for truly private matters that relate
+  uniquely to you (e.g., you need to notify the course staff of a family/medical
+  emergency, etc.).
+
+  Of course, when you run into an issue on an assignment, it may seem like
+  "it applies only to you" because it relates to your code. However,
+  it's possible tha other students will run into the exact same issue
+  you ran into. So, questions about coursework, course logistics, etc.
+  do not qualify as "private matters", and should typically be
+  asked publicly. This way, everyone can benefit from the
+  answer to the question and, if someone runs into the same issue you
+  do, course staff can refer them to the answer they provided in your post.
+
+- **The more information, the better**. We know that, in some
+  classes, you may feel the need to be brief and concise when asking
+  for help to “avoid wasting the professor’s time” or because
+  “professors don’t have time to read long e-mails”. The absolute
+  opposite is true in most CS classes: when asking for help, we prefer
+  that you *overwhelm* us with information.
+
+  In particular, it will be much easier for course staff to help you if they are
+  able to reproduce the exact issue you are encountering (i.e., when
+  they run your code, they must be able to observe the exact same issue
+  you’re encountering). And to do so, they need as much information as
+  possible from you:
+
+   -  If your question relates to your code, and your class uses GitHub
```

Comment:
I think the Github point applies to any class that uses version control. Perhaps we could generalize it.

Reply:
I also think we should change the title if we add the "getting help with this guide" section to the about page. Beyond that, I only have a small comment in the the asking questions page.

Reply:
@vaargueta: Anne just committed some changes to generalize the Git language. Does it look good to you? 

Also, my plan is to add a note saying something like this at the top of this page (after both this PR and the About PR are merged):

> This page provides some useful guidelines on how to ask questions in your classes (and more specifically on an online discussion board). If you have questions about the guide itself, please see the "Getting Help" page in the "About This Guide" section.

Do you think that would be enough?

Reply:
Yes, I think that addresses my concern. I think the note at the top of the page should be enough. If you want it to be particularly visible, I'd consider putting it in a note block just to draw the eye to it.